packages/yew/src/functional/hooks/use_reducer.rs

```diff
@@ -224,7 +224,12 @@ where
                             let should_render_fn = should_render_fn.clone();
                             let mut val = val.borrow_mut();
                             let next_val = (*val).clone().reduce(action);
-                            let should_render = should_render_fn(&next_val, &val);
+
+                            // Check if the reduce action just returned the same `Rc` again
+                            // instead of producing a new one.
+                            let rc_was_reused = Rc::ptr_eq(&val, &next_val);
+
+                            let should_render = !rc_was_reused && should_render_fn(&next_val, &val);
```

Comment:
`use_reducer` currently makes the explicit promise that "This hook will always trigger a re-render upon receiving an action". Please change the docs, as that's a breaking change. If you are reading this from the future and relying on this behaviour, `use_force_update` provides a hook to trigger this rerender manually.

Since `use_reducer_base` is internal, could you push the changes to the callers into their respective `should_render_fn` for clarity in `use_reducer` and `use_reducer_eq` below?